master/internal/resourcemanagers/scaling.go

```diff
@@ -14,22 +17,37 @@ func calculateDesiredNewInstanceNum(taskList *taskList, slotsPerInstance int) in
 			// If a task is already allocated, skip it.
 			continue
 		case it.value().SlotsNeeded == 0:
-			zeroSlotTasks = true
-		case slotsPerInstance == 0:
+			zeroSlotTasks++
+			allTasks++
+		case slotsPerAgent == 0:
 			continue
-		case it.value().SlotsNeeded <= slotsPerInstance:
+		case it.value().SlotsNeeded <= slotsPerAgent:
 			slotSum += it.value().SlotsNeeded
-		case it.value().SlotsNeeded%slotsPerInstance == 0:
+			allTasks++
+		case it.value().SlotsNeeded%slotsPerAgent == 0:
 			slotSum += it.value().SlotsNeeded
+			allTasks++
 		}
 	}
 
+	num1, num2 := 0, 0
```

Comment:
I find this section of the code unclear. Could you pick more descriptive names for the variables?

Reply:
fixed to be `numAgentByZeroSlot` and `numAgentBySlot`